src/lu.jl

```diff
@@ -102,6 +102,39 @@ function lu!(A::HermOrSym{T}, pivot::Union{RowMaximum,NoPivot,RowNonZero} = lupi
     end
     lu!(A.data, pivot; check, allowsingular)
 end
+
+#reusing LU object
+#lu!(F::LU,A) should be dispatched on the type of matrix stored in the LU factorization.
+#but special care needs to be done in the HermOrSym case
+function _lu_copy!(A,x)
+    copyto!(A, x)
+end
+
+function lu_copy!(A,x::HermOrSym)
+    copytri!(A.data, x.uplo, isa(x, Hermitian))
+    @inbounds if isa(x, Hermitian) # realify diagonal
+        for i in axes(x, 1)
+            A[i,i] = x[i,i]
+        end
+    end
+    return A
+end
+
+function lu!(F::LU{<:Any,<:StridedMatrix{<:BlasFloat}}, A; check::Bool = true, allowsingular::Bool = false)
+    lu_copy!(F.factors,A)
```

Comment:
I'm unclear on why `copy!(F.factors, A)` doesn't work here.

Reply:
some code above does this to speed up the `copyto!(A,B::Union{Hermitian,Symmetric})` case, but you are right, this should be optimized at the `copyto!` level, not here.

Reply:
seems like there is already a specialized `copyto!` for this case, introduced here: https://github.com/JuliaLang/LinearAlgebra.jl/commit/c9c36266bab698f43e345645662a6f4a564b8494